packages/ui-react/src/containers/AccountModal/forms/EditPassword.tsx

```diff
@@ -32,24 +32,23 @@ const ResetPassword = ({ type }: { type?: 'add' }) => {
 
       return setSubmitting(false);
     }
-    let resetToken = resetPasswordTokenParam;
-    if (resetPasswordToken) {
-      resetToken = resetPasswordToken;
-    }
+
+    const resetToken = resetPasswordToken || resetPasswordTokenParam;
+
     try {
-      if (user && !resetToken) {
-        await accountController.changePasswordWithOldPassword(oldPassword || '', password, passwordConfirmation);
+      if (resetToken) {
+        await accountController.changePasswordWithToken(emailParam || '', password, resetToken, passwordConfirmation);
       } else {
-        if (!resetToken) {
+        if (!user) {
           setErrors({ form: t('reset.invalid_link') });
-
           return setSubmitting(false);
         }
-        await accountController.changePasswordWithToken(emailParam || '', password, resetToken, passwordConfirmation);
+
+        await accountController.changePasswordWithOldPassword(oldPassword || '', password, passwordConfirmation);
```

Comment:
Could you also try to change the password for JWP within your account? I see infinite screen loader now.